server/content-repo/content-storage.js

```diff
@@ -0,0 +1,88 @@
+'use strict';
+
+const { storage } = require('../config');
+const Promise = require('bluebird');
+const createStorage = require('../common/storage');
+const values = require('lodash/values');
+
+class ContentStorage {
+  constructor(options = {}) {
+    this.storage = createStorage(options);
+    this.options = options;
+  }
+
+  getRepository(repoId) {
+    const key = `${this._getRepoLocation(repoId)}/index.json`;
+    return this.storage.getItem(key);
+  }
+
+  getCatalog() {
+    const key = `${this.options.publishedContentLocation}/index.json`;
+    return this.storage.getItem(key);
+  }
+
+  getContainer(repoId, id, programId) {
+    const key =
+      `${this._getRepoLocation(repoId, programId)}/${id}.container.json`;
+    return this.storage.getItem(key);
+  }
+
+  getExam(repoId, id, programId) {
+    const key = `${this._getRepoLocation(repoId, programId)}/${id}.exam.json`;
+    return this.storage.getItem(key);
+  }
+
+  getAssessments(repoId, id, programId) {
+    const key =
+      `${this._getRepoLocation(repoId, programId)}/${id}.assessments.json`;
+    return this.storage.getItem(key);
+  }
+
+  resolveContainer(container) {
+    return Promise.each(container.elements, element => {
+      return this._resolveElement(element);
+    }).then(() => container);
+  }
+
+  importRepo(programId, repoId) {
+    const src = `${this._getRepoLocation(repoId)}/`;
+    const dest = `${this._getRepoLocation(repoId, programId)}/`;
+    return this.storage.copyDir(src, dest)
+      .then(() => this.getRepository(repoId));
+  }
+
+  _getRepoLocation(repoId, programId) {
+    return programId
+      ? `${this.options.importedContentLocation}/${programId}/${repoId}`
```

Comment:
Since this is within ContentStorage service consider shortening location params to importPath and sourcePath.